graphics/svg_surface.cpp

```diff
@@ -0,0 +1,67 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "graphics/svg_surface.h"
+
+#include "common/endian.h"
+#include "common/stream.h"
+#include "common/textconsole.h"
+#include "graphics/pixelformat.h"
+#define NANOSVG_IMPLEMENTATION
+#include "graphics/nanosvg/nanosvg.h"
+#define NANOSVGRAST_IMPLEMENTATION
+#include "graphics/nanosvg/nanosvgrast.h"
+
+#ifdef SCUMM_BIG_ENDIAN
+#define PIXELFORMAT Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0)
+#else
+#define PIXELFORMAT Graphics::PixelFormat(4, 8, 8, 8, 8, 0, 8, 16, 24)
+#endif
+
+Graphics::SVGSurface::SVGSurface(Common::SeekableReadStream *in, int dw, int dh)
+	: ManagedSurface(dw, dh, PIXELFORMAT) {
+	if (dw == 0 || dh == 0)
+		return;
+
+	int64 size = in->size();
+	char *data = new char[size + 1];
+
+	in->read(data, size);
+	data[size] = '\0';
+
+	NSVGimage *svg = nsvgParse(data, "px", 96);
```

Comment:
Where does the cache go? Why? It is used extensively by our GUI code, and re-rendering it every time is not optimal, particularly in the icons list.

Rendering SVG is very expensive because it uses double-precision FPU arithmetics.

Reply:
Indeed. However from what I could see from the profiling, this bitmap is not reused. So while it looks good on the first sight, in reality this cache didn't provide any speedup, on the contrary.

As written in the other comment, I physically measured the speed this brought and it was enormous.

Reply:
I guess your discomfort with this change is related to the debate I had with @lephilousophe earlier -- whether it's acceptable to narrow SVGBitmap's features to its current usage (load svg file, render, copy and trash) or to keep it as generic as possible.

As the former approach brought significant speedup and simplified other parts of ScummVM code, too it looked like a good idea to go ahead.